### ISSUE TITLE
Double clicking Modal backdrop causes a TypeError

### ISSUE THREAD

**Issue by aripekkako** [reactstrap/reactstrap#673]
### Issue description

- components: `Modal`
- reactstrap version `5.0.0-alpha.3`
- import method `es` (webpack, create-react-app)
- react version `16.0.0`
- bootstrap version `4.0.0-beta.2`

### What is happening?

Double clicking modal backdrop causes a TypeError.

### What should be happening?

Double clicking should close the modal without errors.

### Steps to reproduce issue

1. Click the button to open the modal.
2. Quickly double-click the backdrop.
3. Click anywhere on the page.
4. See an error in the console. Clicks do not work anymore.

### Error message in console

```
Uncaught TypeError: Cannot read property 'contains' of null
    at Modal.handleBackdropClick (reactstrap.es.js:3734)
    at HTMLUnknownElement.callCallback (react-dom.development.js:1299)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:1338)
    at Object.invokeGuardedCallback (react-dom.development.js:1195)
    at Object.invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:1209)
    at executeDispatch (react-dom.development.js:1432)
    at Object.executeDispatchesInOrder (react-dom.development.js:1454)
    at executeDispatchesAndRelease (react-dom.development.js:1969)
    at executeDispatchesAndReleaseTopLevel (react-dom.development.js:1980)
    at Array.forEach (<anonymous>)
    at forEachAccumulated (react-dom.development.js:1946)
    at Object.processEventQueue (react-dom.development.js:2139)
    at runEventQueueInBatch (react-dom.development.js:2151)
    at Object.handleTopLevel [as _handleTopLevel] (react-dom.development.js:2161)
    at handleTopLevelImpl (react-dom.development.js:1800)
    at batchedUpdates (react-dom.development.js:13238)
    at performFiberBatchedUpdates (react-dom.development.js:1646)
    at stackBatchedUpdates (react-dom.development.js:1637)
    at batchedUpdates (react-dom.development.js:1651)
    at Object.batchedUpdatesWithControlledComponents [as batchedUpdates] (react-dom.development.js:1664)
    at dispatchEvent (react-dom.development.js:1874)
```

### Code

https://codepen.io/anon/pen/KygQOz

**Comment by aripekkako** [comment]
This actually happens also if you double click ESC. 

I tried to create a fix the issue, but was not yet successful.

**Comment by TheSharpieOne** [comment]
toggling the modal quickly triggers this. This is related to #360

**Comment by TheSharpieOne** [comment]
Fixed in 2cdf225
